File: F/888-fair-candy-swap/main.go
package main

import "fmt"

func fairCandySwap(A []int, B []int) []int {
	MB := make(map[int]int)

	sum := 0
	suma := 0
	for i := 0; i < len(A); i++ {
		sum += A[i]
		suma += A[i]
	}
	for i := 0; i < len(B); i++ {
		sum += B[i]
		MB[B[i]]++
	}

	half := sum / 2
	for i := 0; i < len(A); i++ {
		_, ok := MB[half-suma+A[i]]
		if ok {
			return []int{A[i], half - suma + A[i]}
		}
	}
	return nil
}

func main() {
	fmt.Println("a")
}

File: F/888-fair-candy-swap/main.rs
struct Solution {}

use std::collections::HashSet;
impl Solution {
  pub fn fair_candy_swap(a: Vec<i32>, b: Vec<i32>) -> Vec<i32> {
    let mut suma = 0;
    let mut sumb = 0;
    let mut m: HashSet<i32> = HashSet::new();
    for i in &a {
      suma += i;
    }
    for i in &b {
      sumb += i;
      m.insert(*i);
    }
    let diff = (sumb - suma) / 2;
    let mut r1 = 0;
    let mut r2 = 0;
    for i in &a {
      let t = i + diff;
      if m.contains(&t) {
        r1 = *i;
        r2 = t;
        break;
      }
    }
    vec![r1, r2]
  }
}

fn main() {
  println!("{}", Solution::fair_candy_swap(vec![0; 5], vec![0; 5]));
}
